node/impl/full/eth.go

```diff
@@ -356,9 +359,15 @@ func (a *EthModule) EthGetTransactionByHashLimited(ctx context.Context, txHash *
 		return nil, nil
 	}
 
-	c, err := a.EthTxHashManager.TransactionHashLookup.GetCidFromHash(*txHash)
+	var c cid.Cid
+	var err error
+	c, err = a.ChainIndexer.GetCidFromHash(ctx, *txHash)
 	if err != nil {
-		log.Debug("could not find transaction hash %s in lookup table", txHash.String())
+		log.Debug("could not find transaction hash %s in chain indexer", txHash.String())
```

Comment:
I've experienced at least one case where the error was more interesting than "not found" but it gets lost in these messages (at the time I recompiled with an inserted error dump).

Consider adding an `ErrNotFound` that you can `errors.Is` here, and elsewhere, to handle differently. Alternatively return a bool 3rd arg as an `ok` which is idiomatic enough to get away with.